src/plugins/plugin.tooltip.js

```diff
@@ -1023,8 +1023,14 @@ export class Tooltip extends Element {
 			}
 		}
 
+		// When there are multiple items shown, but the tooltip position is nearest mode
+		// an update may need to be made because our position may have changed even though
+		// the items are the same as before.
+		const position = positioners[options.position].call(me, active, e);
```

Comment:
For performance reasons, I think we'd probably only want to do this when in nearest mode

Reply:
This is independent of the interaction mode, the active elements are already determined. Here we are determining where to poistion the tooltip considering the active elements and current mouse position etc.

Reply:
That was my thinking. We allow new modes to be added so by checking if the position needs to change, we handle any user added modes.

Perhaps we could pass the computed position down through the update rather than recomputing it again later?

Reply:
> Perhaps we could pass the computed position down through the update rather than recomputing it again later?

That sounds reasonable. The animation target update part can't use the pre-computed value though, because the elements could have moved due animation.

Reply:
I can update this to pass down the computed position if we're all OK with it

Reply:
I'm not quite sure I understand when you say pass it down because it doesn't look like this method is called by `Tooltip.update`. Do you mean you'd set a class member with this value in `update` and then read it here? If so, that sounds fine to me

Reply:
I had meant to pass it down to `Tooltip.update()` since it's recomputed on https://github.com/chartjs/Chart.js/blob/master/src/plugins/plugin.tooltip.js#L639

Reply:
Where's the initial place it's computed?

Reply:
The linked spot used to be the first place it was computed. This MR now computes it before trying to dispatch the update call

Reply:
Ok, yeah passing it to `update` sounds good to me.

Somewhat unrelated, but a lot of these methods could probably be made private. It looks like `update` and most of the other methods are public, but only called within this file. I had expected `update` to be called from outside the class since it was public, which threw me off a bit and made it take me a bit longer to understand this stuff.

Reply:
Got into making that change and it seems to break a fair number of tests. Not sure if it's worth it since the performance gain is minimal at best

Reply:
Yeah, that's fine with me. I'll resolve this comment